commands.js

```diff
@@ -286,6 +287,50 @@ var commands = exports.commands = {
 	},
 	makechatroomhelp: ["/makechatroom [roomname] - Creates a new room named [roomname]. Requires: ~"],
 
+	personalroom: 'groupchat',
+	groupchat: function (target, room, user) {
+		// First and foremost, check the number of personal rooms owned by the user.
+		if (user.personalRooms && user.personalRooms >= MAX_PERSONAL_ROOMS) return this.sendReply("You can't own more than " + MAX_PERSONAL_ROOMS + " personal rooms.");
+
+		// Chosen name have the same limitations as regular rooms.
+		if (target.includes(',') || target.includes('|') || target.includes('[') || target.includes('-')) {
+			return this.sendReply("Room titles can't contain any of: ,|[-");
+		}
+
+		// Check if the custom part of the room title is empty. It needs a title.
+		var id = toId(target);
+		// If there isn't any, we roll our own random number title.
+		if (!id) {
+			target = Math.floor(Math.random() * 10000);
+		}
+
+		// Create the name of the room using the chosen name plus the groupchat- namespace.
+		var roomName = "groupchat-" + target;
+		id = toId(roomName);
+
+		// Personal rooms can always be created, but the name must still be unique.
+		if (Rooms.search(id)) return this.sendReply("The room '" + roomName + "' already exists.");
+		if (Rooms.global.addChatRoom(roomName, true)) {
+			var targetRoom = Rooms.search(roomName);
+			// Make the room private, as it can't be public.
+			targetRoom.isPrivate = true;
+			targetRoom.chatRoomData.isPrivate = true;
+			// The room is modjoin by default. This shouldn't be changed.
+			targetRoom.modjoin = true;
```

Comment:
Hm, if it's to be treated like a group PM, then drivers+ probably shouldn't be able to see/moderate it (i.e. it should be private instead of hidden)...